servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1094,4 +1120,30 @@ public void reportLostFile(long fileId) throws FileDoesNotExistException {
       return mDependencyMap.getPriorityDependencyList();
     }
   }
+
+  /**
+   * MasterInodeTTL periodic check.
+   */
+  public final class MasterInodeTTLCheckExecutor implements HeartbeatExecutor {
+    @Override
+    public void heartbeat()  {
+      synchronized (mInodeTree) {
+        List<Inode> inodes = mInodeTree.getInodeChildrenRecursive(mInodeTree.getRoot());
```

Comment:
You may be able to do something by maintaining a separate data structure specifically for TTLed inodes, but we can take a look at it separately (Ticket-1044)